---
title: "Predicting malarial signal <br> peptides with signalHsmm"
author: "Micha≈Ç Burdukiewicz"
date: "25-04-2016"
output:
  ioslides_presentation: null
  fig_height: 5.5
  beamer_presentation: default
slidy_presentation: default
fig_width: 9.5
widescreen: yes
---
  
```{r, echo = FALSE, message = FALSE, results='asis',warning=FALSE}
library(DT)
library(knitr)
require(XML)
require(seqinr)
require(fitdistrplus)
require(dplyr)
require(signalHsmm)
require(hmeasure)
require(pbapply)
require(reshape2)
require(hmeasure)
require(xtable)
require(biogram)
require(ggplot2)
require(grid)
require(gridExtra)

options(DT.options = list(iDisplayLength = 5, searching = TRUE))

deg_freq_plot <- function(x) {
  agg_signals <- group_by(x, only_sig, aa) %>% 
    summarise(value = mean(value)) %>% 
    ungroup
  
  ggplot(agg_signals, aes(x = aa, y = value, fill = !only_sig)) +
    geom_bar(stat = "identity", position = "dodge") +
    scale_x_discrete("Amino acid") +
    scale_y_continuous("Mean frequency") +
    scale_fill_discrete("Recognized by signalP") +
    my_theme
}

load("presentation.RData")

```

## signalHsmm versus other software

signalHsmm has the most sensitive predictions of malarial signal peptides.

```{r, echo = FALSE, message = FALSE, results='asis',warning=FALSE}
datatable(all_TP, rownames = FALSE, filter = "top")
```

The usage of reduced amino acid alphabet has an impact on the sensitivity.

## Lengths of signal peptides

```{r, echo = FALSE, message = FALSE, results='asis',warning=FALSE}
ggplot(cs_dat, aes(x = !only_sig, y = cs)) +
  geom_dotplot(binaxis = "y", stackdir = "center", binwidth = 0.8) +
  scale_x_discrete("Recognized by signalP") +
  scale_y_continuous("Position of cleavage site") +
  my_theme
```

## Frequency - full alphabet

```{r, echo = FALSE, message = FALSE, results='asis',warning=FALSE}
deg_freq_plot(signals)
```

## Frequency - reduced alphabet

```{r, echo = FALSE, message = FALSE, results='asis',warning=FALSE}
deg_freq_plot(signals_deg)
```

## Problematic proteins

 - Merozoite surface antigen 2 (Plasmodium falciparum (isolate 3D7))
 - Apical membrane antigen 1 (Plasmodium fragile)
 - Histidine-rich protein PFHRP-III (Plasmodium falciparum)
 - Sporozoite surface protein P36p (Plasmodium falciparum (isolate 3D7))
 - Sporozoite surface protein P36 (Plasmodium berghei (strain Anka))
 - Sporozoite surface protein P36 (Plasmodium falciparum (isolate 3D7))
 
## Protein table

| Name                             | Source                              | Evidence                                   |
|----------------------------------|-------------------------------------|--------------------------------------------|
| Merozoite surface antigen 2      | Plasmodium falciparum (isolate 3D7) | Experimental evidence at protein leveli    |
| Apical membrane antigen 1        | Plasmodium fragile                  | Protein inferred from homology             |
| Histidine-rich protein PFHRP-III | Plasmodium falciparum               | Experimental evidence at transcript level  |
| Sporozoite surface protein P36p  | Plasmodium falciparum (isolate 3D7) | Protein inferred from homology             |
| Sporozoite surface protein P36   | Plasmodium berghei (strain Anka)    | Experimental evidence at transcript level  |
| Sporozoite surface protein P36   | Plasmodium falciparum (isolate 3D7) | Protein inferred from homology             |